### PR TITLE
refactor(semantic): flatbuffers for basic types

### PR DESCRIPTION
**Pull request by jlapacik**
### Done checklist
- [ ] docs/SPEC.md updated
- [ ] Test cases written

Initialize flatbuffers for basic types

### CODE REVIEW

File: semantic/monotype.go [hunk 1]
```diff
@@ -103,7 +103,15 @@ func (mt MonoType) Kind() Kind {
 }
 
 var (
-	basicTable *flatbuffers.Table
+	tableBool     *flatbuffers.Table
```
Comment (jsternberg) [inline semantic/monotype.go]: Why are these necessary?

File: semantic/monotype.go [hunk 2]
```diff
@@ -117,20 +125,133 @@ var (
...
 
 func init() {
-	builder := flatbuffers.NewBuilder(1024)
-	fbsemantic.BasicStart(builder)
-	fbsemantic.BasicAddT(builder, fbsemantic.TypeBool)
-	basicBool := fbsemantic.BasicEnd(builder)
 
-	// TODO (algow): initial all basic values
+	{
+		builder := flatbuffers.NewBuilder(0)
```
Comment (jsternberg) [inline semantic/monotype.go]: It seems like the below code can be refactored into a single function that returns the monotype. There's only one parameter that changes and that's the type.

File: semantic/monotype.go (hunk 1 above)
Comment (jlapacik) [reply semantic/monotype.go]: Not entirely sure.

File: semantic/monotype.go [hunk 3]
```diff
@@ -117,20 +115,32 @@ var (
...
-	basicTable = &flatbuffers.Table{
+	fbsemantic.BasicStart(builder)
+	fbsemantic.BasicAddT(builder, ty)
+	offset := fbsemantic.BasicEnd(builder)
+
+	builder.Finish(offset)
+	table := flatbuffers.Table{
 		Bytes: builder.FinishedBytes(),
-		Pos:   basicBool,
+		Pos:   offset,
```
Comment (jsternberg) [inline semantic/monotype.go]: I don't think this is correct. I think the proper code for this is:

```
table := fbsemantic.GetRootAsBasic(builder.FinishedBytes(), 0).Table()
basic, _ := NewMonoType(&table, fbsemantic.MonoTypeBasic)
```

File: semantic/monotype.go [hunk 4]
```diff
@@ -117,20 +115,32 @@ var (
...
+	BasicFloat = buildBasic(fbsemantic.TypeFloat)
+	BasicString = buildBasic(fbsemantic.TypeString)
+	BasicDuration = buildBasic(fbsemantic.TypeDuration)
+	BasicTime = buildBasic(fbsemantic.TypeTime)
+	BasicRegexp = buildBasic(fbsemantic.TypeRegexp)
+	BasicBytes = buildBasic(fbsemantic.TypeBytes)
+}
 
-	// TODO (algow): initial all basic values
+func buildBasic(ty fbsemantic.Type) MonoType {
```
Comment (jsternberg) [inline semantic/monotype.go]: Nitpick: `buildBasicType`

File: semantic/monotype.go [hunk 5]
```diff
@@ -117,20 +115,32 @@ var (
...
+
+	builder.Finish(offset)
+	table := flatbuffers.Table{
 		Bytes: builder.FinishedBytes(),
-		Pos:   basicBool,
+		Pos:   offset,
 	}
-	BasicBool, _ = NewMonoType(basicTable, fbsemantic.MonoTypeBasic)
+
+	basic, _ := NewMonoType(&table, fbsemantic.MonoTypeBasic)
```
Comment (jsternberg) [inline semantic/monotype.go]: Can you catch the error here and panic if it does?

**Comment by jsternberg** [comment]
Can you also add test cases for each of these? I think you should be able to use the `String()` method as @wolffcm advised.

**Comment by jsternberg** [comment]
Not needed anymore.
